sdk/test/trace/sampler_benchmark.cc

```diff
@@ -0,0 +1,132 @@
+#include "opentelemetry/sdk/trace/samplers/always_off.h"
```

Comment:
From discussions with Johannes, he mentioned wanting to optimize the No-op Span use case. Can you add benchmarking for this case?

Reply:
That's tested from line 213 to 231. I created a tracer with an always off sampler, which then creates a no-op span every time StartSpan is called. Is my understand of what I should be testing incorrect?